### DISCUSSION TITLE
How to prevent automatically posting comments with or without user sign in

### DISCUSSION THREAD

**Discussion by Suleman-Elahi** [walinejs/waline#1831] [Q&A] [answered]
Is there a way in waline to make all the comments go to waiting list irrespective of if they come from a registered user or anonymous user. And then admins can manually approve them to add to the site? I have a huge website and get a lot of spam on disqus but want to move to waline. That's why I am asking this.

**Comment by lizheming** [comment, marked as answer]
https://waline.js.org/en/reference/server/env.html#safty try `COMMENT_AUDIT` env~

**Comment by Suleman-Elahi** [comment]
Okay thanks. Can you also tell me a good way to convert Disqus XML to Waline SQL ... I tried the Migration tool on the website but my XML file is very large 32MB and has more than 100K comments... and then tool is hanging. It can't generate the SQL.

**Reply by lizheming** [reply]
Maybe you can wait more time? It's handling data when page is hanging 🥲 It will generate a csv file.
Or you can copy migration tool's transform code to a node script file to handle it in your localhost?


There have `disqus2lc` and `lc2csv` two functions, you can copy to your code.
https://github.com/walinejs/waline/blob/main/docs/src/.vuepress/components/transform/disqus2lc.js
https://github.com/walinejs/waline/blob/main/docs/src/.vuepress/components/transform/lc2csv.js
```js
// migrate.js

// ... copy function `disqus2lc` and `lc2csv` code to here

const fs = require('fs');
const disqus = fs.readFileSync('./disqus.xml', 'utf8');
const output = lc2csv(disqus2lc(disqus));
fs.writeFileSync(output, 'output.csv', 'utf8');
```

**Comment by Suleman-Elahi** [comment]
Ok. I will try and report baack :)

**Comment by Suleman-Elahi** [comment]
Hi,

I tried and I am getting this and don't know to how to resolve... I haven't touched Node.js before..so I put together from Google and your snippet.

```
PS C:\Users\sul00\Desktop\wg> node .\migrate.js
C:\Users\sul00\Desktop\wg\migrate.js:6
  const posts = Array.from(dom.querySelectorAll('post')).filter((postEl) => {
                               ^
TypeError: dom.querySelectorAll is not a function
    at disqus2lc (C:\Users\sul00\Desktop\wg\migrate.js:6:32)
    at Object.<anonymous> (C:\Users\sul00\Desktop\wg\migrate.js:176:23)
    at Module._compile (node:internal/modules/cjs/loader:1112:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47

Node.js v18.4.0
PS C:\Users\sul00\Desktop\wg>
```
Here's the full script: https://paste.ee/p/MDYMW

I tried with `xmldom` as well as `@xmldom/xmldom` , both giving the same error.
